api/src/pcapi/core/external/attributes/api.py

```diff
@@ -223,6 +234,29 @@ def get_pro_attributes(email: str) -> models.ProAttributes:
     )
 
 
+def _check_if_pro_attribute_has_collective_offers(offerers, attributes):
+    for offerer in offerers:
+        venues = offerer.managedVenues
+        for venue in venues:
+            if venue.collectiveOffers:
+                for collective_offer in venue.collectiveOffers:
+                    if collective_offer.isActive or collective_offer.isSoldOut:
+                        attributes.update(
+                            {
+                                "has_collective_offers": True,
+                            }
+                        )
+            elif venue.collectiveOfferTemplates:
+                for collective_offer_template in venue.collectiveOfferTemplates:
+                    if collective_offer_template.isActive or collective_offer_template.isSoldOut:
```

Comment:
(style / question de goût) Pour remplacer ce genre de boucle for+if, tu peux utiliser `any` : 

```python
if any(template.isActive or template.isSoldOut for template in venue.collectiveOfferTemplates):
    attributes["has_collective_offers"] = True
```

`any()` permet de ne pas parcourir toute une liste (ou générateur, ici).

Reply:
wouah interressant, merci !